server/index.js

```diff
@@ -165,6 +164,48 @@ app.post('/api/cart', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.post('/api/orders', (req, res, next) => {
+
+  let sessionCartId = req.session.cartId;
+  const paymentName = req.body.name;
+  const paymentCreditCard = req.body.creditCard;
+  const paymentShippingAddress = req.body.shippingAddress;
+
+  if (!req.session.cartId) {
+    return res.status(400).json({
+      error: 'no session cartId found!'
+    });
+  }
+
+  if (!paymentName || !paymentCreditCard || !paymentShippingAddress) {
+    res.status(402).json({
+      error: 'payment information required!'
+    });
+  } else {
+
+    const insert = `
+      insert into "orders" ("cartId", "name", "creditCard", "shippingAddress")
+      values ($1, $2, $3, $4)
+      returning *
+    `;
+
+    const values = [sessionCartId, paymentName, paymentCreditCard, paymentShippingAddress];
+
+    db.query(insert, values)
+      .then(result => {
+        if (result.rows[0]) {
+          sessionCartId = null;
+          res.status(201).json(result.rows[0]);
+        } else {
+          res.status(500).json({
+            error: 'an unexpected error occurred'
+          });
+        }
```

Comment:
If an insert doesn't throw an error, and you use `returning`, then the row will never be falsy, so you can remove the conditional logic.

Also, we need to actually `delete req.session.cartId` instead of nulling a variable.